server/core/store_stats.go

```diff
@@ -0,0 +1,156 @@
+	ss.avgMaxAvailableDeviation.Add(ss.maxAvailableDeviation.Get())
+}
+
+// GetStoreStats returns the statistics information of the store.
+func (ss *storeStats) GetStoreStats() *pdpb.StoreStats {
+	ss.mu.RLock()
+	defer ss.mu.RUnlock()
+	return ss.rawStats
+}
+
+// GetCapacity returns the capacity size of the store.
+func (ss *storeStats) GetCapacity() uint64 {
+	ss.mu.RLock()
+	defer ss.mu.RUnlock()
+	return ss.rawStats.GetCapacity()
+}
+
+// GetAvailable returns the available size of the store.
+func (ss *storeStats) GetAvailable() uint64 {
+	ss.mu.RLock()
+	defer ss.mu.RUnlock()
+	return ss.rawStats.GetAvailable()
+}
+
+// GetUsedSize returns the used size of the store.
+func (ss *storeStats) GetUsedSize() uint64 {
+	ss.mu.RLock()
+	defer ss.mu.RUnlock()
+	return ss.rawStats.GetUsedSize()
+}
+
+// GetBytesWritten returns the bytes written for the store during this period.
+func (ss *storeStats) GetBytesWritten() uint64 {
+	ss.mu.RLock()
+	defer ss.mu.RUnlock()
+	return ss.rawStats.GetBytesWritten()
+}
+
+// GetBytesRead returns the bytes read for the store during this period.
+func (ss *storeStats) GetBytesRead() uint64 {
+	ss.mu.RLock()
+	defer ss.mu.RUnlock()
+	return ss.rawStats.GetBytesRead()
+}
+
+// GetKeysWritten returns the keys written for the store during this period.
+func (ss *storeStats) GetKeysWritten() uint64 {
+	ss.mu.RLock()
+	defer ss.mu.RUnlock()
+	return ss.rawStats.GetKeysWritten()
+}
+
+// GetKeysRead returns the keys read for the store during this period.
+func (ss *storeStats) GetKeysRead() uint64 {
+	ss.mu.RLock()
+	defer ss.mu.RUnlock()
+	return ss.rawStats.GetKeysRead()
+}
+
+// IsBusy returns if the store is busy.
+func (ss *storeStats) IsBusy() bool {
+	ss.mu.RLock()
+	defer ss.mu.RUnlock()
+	return ss.rawStats.GetIsBusy()
+}
+
+// GetSendingSnapCount returns the current sending snapshot count of the store.
+func (ss *storeStats) GetSendingSnapCount() uint32 {
+	ss.mu.RLock()
+	defer ss.mu.RUnlock()
+	return ss.rawStats.GetSendingSnapCount()
+}
+
+// GetReceivingSnapCount returns the current receiving snapshot count of the store.
+func (ss *storeStats) GetReceivingSnapCount() uint32 {
+	ss.mu.RLock()
+	defer ss.mu.RUnlock()
+	return ss.rawStats.GetReceivingSnapCount()
+}
+
+// GetApplyingSnapCount returns the current applying snapshot count of the store.
+func (ss *storeStats) GetApplyingSnapCount() uint32 {
+	ss.mu.RLock()
+	defer ss.mu.RUnlock()
+	return ss.rawStats.GetApplyingSnapCount()
+}
+
+// GetAvgAvailable returns available size after the spike changes has been smoothed.
+func (ss *storeStats) GetAvgAvailable() uint64 {
+	ss.mu.RLock()
+	defer ss.mu.RUnlock()
+	return climp0(ss.avgAvailable.Get())
+}
+
+// GetAvailableDeviation returns approximate magnitude of available in the recent period.
+func (ss *storeStats) GetAvailableDeviation() uint64 {
+	ss.mu.RLock()
+	defer ss.mu.RUnlock()
+	return climp0(ss.avgMaxAvailableDeviation.Get())
+}
+
+func climp0(v float64) uint64 {
```

Comment:
What does climp0 here mean?

Reply:
no special meaning 😢  do you have better idea?

Reply:
Will the float be truncated?

Reply:
Yes. It is usually a very large value (unit is byte) so the truncation won't matter.